react/src/lib/components/Menu/components/MenuDrawer/MenuDrawer.tsx

```diff
@@ -0,0 +1,202 @@
+import React from "react";
+import PropTypes from "prop-types";
+import useSize from "@react-hook/size";
+
+import { MenuDrawerPosition } from "../../types/menu-position";
+
+import "./MenuDrawer.css";
+
+type MenuDrawerProps = {
+    position: MenuDrawerPosition;
+    open: boolean;
+    children?: React.ReactNode;
+};
+
+type Position = {
+    left: number | "auto";
+    top: number | "auto";
+    right: number | "auto";
+    bottom: number | "auto";
+};
+
+export const MenuDrawer = React.forwardRef<HTMLDivElement, MenuDrawerProps>(
+    (props, ref) => {
+        const [position, setPosition] = React.useState<Position>({
+            left: 0,
+            top: 0,
+            right: 0,
+            bottom: 0,
+        });
+        const [visible, setVisible] = React.useState<boolean>(false);
+
+        const drawerRef =
+            (ref as React.RefObject<HTMLDivElement>) ||
+            React.useRef<HTMLDivElement>(null);
+        const drawerWidth = useSize(drawerRef)[0];
+
+        const slideInDrawer = React.useCallback(
+            (pos: Position) => {
+                if (props.position === "left") {
+                    let currentPosition = pos.left as number;
+                    const interval = setInterval(() => {
```

Comment:
Should not be problematic the way it is implemented now since it is self-contained and `setPosition` gets overriden. However, it's cleaner to add a `Ref` and clear it on unmount. Valid point :+1: